File: core-platform/apps/core-webapp/src/services/clientRepresentative.service.ts
import axiosInstance from '../axiosInstance';
import type { ClientRepresentativeDTO } from '../types/client.types';

const CLIENT_REP_API_BASE = '/client-representatives';

export const clientRepresentativeService = {
  // Add a representative
  addRepresentative: async (data: Omit<ClientRepresentativeDTO, 'id'>): Promise<ClientRepresentativeDTO> => {
    const response = await axiosInstance.post(CLIENT_REP_API_BASE, data);
    return response.data.data;
  },

  // Update a representative
  updateRepresentative: async (
    id: number,
    data: Partial<ClientRepresentativeDTO>
  ): Promise<ClientRepresentativeDTO> => {
    const response = await axiosInstance.put(`${CLIENT_REP_API_BASE}/${id}`, data);
    return response.data.data;
  },

  // Activate a representative
  activateRepresentative: async (id: number): Promise<void> => {
    await axiosInstance.patch(`${CLIENT_REP_API_BASE}/${id}/activate`);
  },

  // Deactivate a representative
  deactivateRepresentative: async (id: number): Promise<void> => {
    await axiosInstance.patch(`${CLIENT_REP_API_BASE}/${id}/deactivate`);
  },

  // Get representative by ID
  getRepresentativeById: async (id: number): Promise<ClientRepresentativeDTO> => {
    const response = await axiosInstance.get(`${CLIENT_REP_API_BASE}/${id}`);
    return response.data.data;
  },

  // Get all representatives for a client
  getRepresentativesByClient: async (clientId: number): Promise<ClientRepresentativeDTO[]> => {
    const response = await axiosInstance.get(`${CLIENT_REP_API_BASE}/client/${clientId}`);
    return response.data.data;
  },

  // Get all representatives for an organization
  getAllRepresentatives: async (organizationId: number): Promise<ClientRepresentativeDTO[]> => {
    const response = await axiosInstance.get(`${CLIENT_REP_API_BASE}/organization/${organizationId}`);
    return response.data.data;
  },

  // Get primary representative for a client
  getPrimaryRepresentative: async (clientId: number): Promise<ClientRepresentativeDTO> => {
    const response = await axiosInstance.get(`${CLIENT_REP_API_BASE}/client/${clientId}/primary`);
    return response.data.data;
  },

  // Check if contact is linked to client
  isContactLinked: async (clientId: number, contactId: number): Promise<boolean> => {
    const response = await axiosInstance.get(`${CLIENT_REP_API_BASE}/exists`, {
      params: { clientId, contactId }
    });
    return response.data.data;
  }
};

export default clientRepresentativeService;

File: core-platform/services/core-service/src/main/java/com/dev/core/service/ClientRepresentativeService.java
package com.dev.core.service;

import com.dev.core.model.ClientRepresentativeDTO;
import java.util.List;

public interface ClientRepresentativeService {

    /**
     * Add a representative (contact) for a given client.
     */
    ClientRepresentativeDTO addRepresentative(ClientRepresentativeDTO dto);

    /**
     * Update representative role or details.
     */
    ClientRepresentativeDTO updateRepresentative(Long id, ClientRepresentativeDTO dto);

    /**
     * Remove (soft delete) a representative.
     */
    void deactivateRepresentative(Long id);

    /**
     * Reactivate representative.
     */
    void activateRepresentative(Long id);

    /**
     * Get representative by ID.
     */
    ClientRepresentativeDTO getRepresentativeById(Long id);

    /**
     * Get all representatives for a given client.
     */
    List<ClientRepresentativeDTO> getRepresentativesByClient(Long clientId);

    /**
     * Get all representatives for an organization.
     */
    List<ClientRepresentativeDTO> getAllRepresentatives(Long organizationId);

    /**
     * Get the primary contact for a client (if exists).
     */
    ClientRepresentativeDTO getPrimaryRepresentative(Long clientId);

    /**
     * Check if a given contact is already linked to a client.
     */
    boolean isContactLinkedToClient(Long clientId, Long contactId);
}
